store/postgres/asset_repository_test.go

```diff
@@ -127,6 +130,104 @@ func (r *AssetRepositoryTestSuite) insertRecord() (assets []asset.Asset) {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			Types: []asset.Type{asset.TypeTable},
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `type IN ($1)`
+		r.Equal(actualQuery, expectedQuery[1])
+	})
+
+	r.Run("should return sql query with services filter", func() {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			Services: []string{"mysql", "kafka"},
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `service IN ($1,$2)`
+		r.Equal(actualQuery, expectedQuery[1])
+	})
+
+	r.Run("should return sql query with query fields filter", func() {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			QueryFields: []string{"name", "description"},
+			Query:       "demo",
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `(name ILIKE $1 OR description ILIKE $2)`
+		r.Equal(actualQuery, expectedQuery[1])
+	})
+
+	r.Run("should return sql query with nested data query filter", func() {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			QueryFields: []string{"data.landscape.properties.project-id", "data.title"},
+			Query:       "columbus_001",
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `(data->'landscape'->'properties'->>'project-id' ILIKE $1 OR data->>'title' ILIKE $2)`
+		r.Equal(actualQuery, expectedQuery[1])
+	})
+
+	r.Run("should return sql query with asset's data fields filter", func() {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			Data: map[string]string{
+				"entity":  "odpf",
+				"country": "th",
+			},
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `data->>'entity' = 'odpf' AND data->>'country' = 'th'`
+		r.Equal(actualQuery, expectedQuery[1])
+	})
+
+	r.Run("should return sql query with asset's nested data fields filter", func() {
+		results := r.repository.BuildFilterQuery(r.builder, asset.Config{
+			Data: map[string]string{
+				"landscape.properties.project-id": "columbus_001",
+				"country":                         "vn",
+			},
+		})
+		query, _, err := results.ToSql()
+		r.Require().NoError(err)
+		query, err = sq.Dollar.ReplacePlaceholders(query)
+		expectedQuery := strings.Split(query, "WHERE ")
+		actualQuery := `data->'landscape'->'properties'->>'project-id' = 'columbus_001' AND data->>'country' = 'vn'`
```

Comment:
wouldn't it be like this?
```suggestion
		expectedQuery := `data->'landscape'->'properties'->>'project-id' = 'columbus_001' AND data->>'country' = 'vn'`
```